packages/http/src/decorators.ts

```diff
@@ -195,96 +208,99 @@ export function $statusCode(context: DecoratorContext, entity: ModelProperty) {
+export function getStatusCodesWithDiagnostics(
+  program: Program,
+  type: ModelProperty
+): [HttpStatusCodes, readonly Diagnostic[]] {
+  return getStatusCodesFromType(program, type, type);
+}
+
+export function getStatusCodes(program: Program, entity: ModelProperty): HttpStatusCodes {
+  return ignoreDiagnostics(getStatusCodesWithDiagnostics(program, entity));
 }
 
 // Reference: https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html
-export function getStatusCodeDescription(statusCode: string) {
-  switch (statusCode) {
-    case "200":
+export function getStatusCodeDescription(statusCode: number | "*" | HttpStatusCodeRange | string) {
+  if (typeof statusCode === "object") {
+    return rangeDescription(statusCode.start, statusCode.end);
+  }
+  const statusCodeNumber = typeof statusCode === "string" ? parseInt(statusCode, 10) : statusCode;
+  switch (statusCodeNumber) {
+    case 200:
       return "The request has succeeded.";
-    case "201":
+    case 201:
       return "The request has succeeded and a new resource has been created as a result.";
-    case "202":
+    case 202:
       return "The request has been accepted for processing, but processing has not yet completed.";
-    case "204":
+    case 204:
       return "There is no content to send for this request, but the headers may be useful. ";
-    case "301":
+    case 301:
       return "The URL of the requested resource has been changed permanently. The new URL is given in the response.";
-    case "304":
+    case 304:
       return "The client has made a conditional request and the resource has not been modified.";
-    case "400":
+    case 400:
       return "The server could not understand the request due to invalid syntax.";
-    case "401":
+    case 401:
       return "Access is unauthorized.";
-    case "403":
+    case 403:
       return "Access is forbidden";
-    case "404":
+    case 404:
       return "The server cannot find the requested resource.";
-    case "409":
+    case 409:
       return "The request conflicts with the current state of the server.";
-    case "412":
+    case 412:
       return "Precondition failed.";
-    case "503":
+    case 503:
       return "Service unavailable.";
   }
 
-  switch (statusCode.charAt(0)) {
-    case "1":
-      return "Informational";
-    case "2":
-      return "Successful";
-    case "3":
-      return "Redirection";
-    case "4":
-      return "Client Error";
-    case "5":
-      return "Server Error";
-  }
+  return rangeDescription(statusCodeNumber, statusCodeNumber);
+}
 
-  // Any valid HTTP status code is covered above.
+function rangeDescription(start: number, end: number) {
+  if (start >= 100 && end <= 199) {
+    return "Informational";
+  } else if (start >= 200 && end <= 299) {
+    return "Successful";
+  } else if (start >= 300 && end <= 399) {
+    return "Redirection";
+  } else if (start >= 400 && end <= 499) {
+    return "Client error";
+  } else if (start >= 500 && end <= 599) {
```

Comment:
Seems like 400 -> 599 could be described as 'Error', but this is a nit.

Reply:
http docs ususually treat those differently 
<img width="373" alt="image" src="https://github.com/microsoft/typespec/assets/1031227/8e8fe207-3ff3-47b4-bade-8395ba9557da">